### PR TITLE
fix 2d stop_on_slope  (issue #21595)

### LINKED ISSUE #21595: stop_on_slope doesn't work as expected (on KinematicBody2D at least)
**Godot version:**
v3.1 alpha, build cb63cc8

**OS/device including version:**
Windows 8.1 x64

**Issue description:**

Recently `move_and_slide` has seen a lot of changes and the way it handles slanted floor was altered.
Previously it used a `slope_stop_min_velocity` value to determine the speed threshold where to stop sliding and it worked as expected.
Now it uses a boolean `stop_on_slope ` value to determine whether to stop sliding on any slope.
It stops the sliding down, yes, but it is also making it difficult to slide up the slopes.

**Steps to reproduce:**
Use `move_and_slide `with `stop_on_slope  ` set to `true `to make a KinematicBody2D slide over slanted terrain.

**Minimal reproduction project:**
[using_kinematic2d.zip](https://github.com/godotengine/godot/files/2335899/using_kinematic2d.zip)

### PR DESCRIPTION
**Pull request by piratesephiroth**
*Bugsquad edit:* Fixes #21595

### CODE REVIEW

File: scene/2d/physics_body_2d.cpp [hunk 1]
```diff
@@ -1271,14 +1271,10 @@ Vector2 KinematicBody2D::move_and_slide(const Vector2 &p_linear_velocity, const
...
 
-				if (p_stop_on_slope && is_on_slope) {
-					motion = motion.slide(p_floor_direction);
-					lv = lv.slide(p_floor_direction);
-				} else {
 					Vector2 n = collision.normal;
 					motion = motion.slide(n);
 					lv = lv.slide(n);
-				}
+
```
Comment (LikeLakers2) [inline scene/2d/physics_body_2d.cpp]: `clang-format` is complaining about too many tabs here (as well as the blank line). https://travis-ci.org/godotengine/godot/jobs/423337098#L533

Comment (piratesephiroth) [reply scene/2d/physics_body_2d.cpp]: fixed

**Comment by akien-mga** [comment]
Rebased and squashed myself as 8d84da0.

**Comment by akien-mga** [comment]
Amended to remove unused variable.

File: scene/2d/physics_body_2d.cpp [hunk 2]
```diff
@@ -1291,31 +1290,24 @@ Vector2 KinematicBody2D::move_and_slide(const Vector2 &p_linear_velocity, const
 						floor_velocity = collision.collider_vel;
 
 						if (p_stop_on_slope) {
-							if (Vector2() == lv_n + p_floor_direction) {
+							if (Vector2() == lv_n + p_floor_direction && collision.travel.length() < 1) {
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply scene/2d/physics_body_2d.cpp]: That would work yeah.

**Pull request merged**